app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreDatabaseServiceTest.kt

```diff
@@ -83,14 +84,20 @@ class FirestoreDatabaseServiceTest : TestCase() {
         }
     }
 
-    @Test(expected = IllegalArgumentException::class)
     fun test_getAllPictures_throwsIfCategoryNotPresent(): Unit = runBlocking {
-        demoInterface.getAllPictures(fakeCategory)
+        try {
+            demoInterface.getAllPictures(fakeCategory)
```

Comment:
This pattern will lead to false negatives. I think JUnit4 doesn't have an `assertThrows` to the dismay of many, but it might be worth briefly looking around. Failing that we can roll our own

Reply:
I did not find any `assertThrows`, this is the best I could come up with...

Reply:
Here's my suggestion then, successfully tested for false negatives

```kotlin
        runCatching {
            demoInterface.getAllPictures(fakeCategory)
        }.fold({
            fail("unexpected successful completion")
        }, {
            assertThat(it, instanceOf(IllegalArgumentException::class.java))
        })
```

Reply:
Thanks, I will implement this after this sprint.

Reply:
ok